arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModesViewController.swift

```diff
@@ -0,0 +1,133 @@
+                return "Manual cache"
+            }
+        }
+        
+        var mode: AGSFeatureRequestMode {
+            switch self {
+            case .cache:
+                return .onInteractionCache
+            case .noCache:
+                return .onInteractionNoCache
+            case .manualCache:
+                return .manualCache
+            }
+        }
+    }
+    
+    /// Prompt mode selection.
+    @IBAction func modeButtonTapped(_ button: UIBarButtonItem) {
+        // Set up action sheets.
+        let alertController = UIAlertController(
+            title: "Choose a feature request mode.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        // Create an action for each mode.
+        FeatureRequestMode.allCases.forEach { mode in
+            let action = UIAlertAction(title: mode.title, style: .default) { [self] _ in
+                changeFeatureRequestMode(to: mode.mode)
+                let message = "\(mode.title) enabled."
+                setStatus(message: message)
+            }
+            alertController.addAction(action)
+        }
+        // Add a cancel action.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = modeBarButtonItem
+        present(alertController, animated: true)
+    }
+    
+    /// Populate for manual cache mode.
+    @IBAction func populateManualCache(_ button: UIBarButtonItem) {
+        // Set query parameters.
+        let params = AGSQueryParameters()
+        // Query for all tree conditions except "dead" with coded value '4'.
+        params.whereClause = "Condition < '4'"
+        // Show the progress HUD.
+        UIApplication.shared.showProgressHUD(message: "Populating")
+        // Populate features based on the query.
+        self.featureTable.populateFromService(with: params, clearCache: true, outFields: ["*"]) { [weak self] (result: AGSFeatureQueryResult?, error: Error?) in
+            guard let self = self else { return }
+            // Hide progress HUD.
+            UIApplication.shared.hideProgressHUD()
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Display the number of features found.
+                let message = "Populated \(result?.featureEnumerator().allObjects.count ?? 0) features."
+                self.setStatus(message: message)
+            }
+        }
+    }
+    
+    /// Set the appropriate feature request mode.
+    private func changeFeatureRequestMode(to mode: AGSFeatureRequestMode) {
+        // Enable or disable the populate bar button item when appropriate.
+        if mode == .manualCache {
+            populateBarButtonItem.isEnabled = true
+        } else {
+            populateBarButtonItem.isEnabled = false
+        }
+        let map = mapView.map
```

Comment:
We may just force unwrap here, to avoid optional chaining twice below.
```suggestion
        let map = mapView.map!
```